modules/platforms/dotnet/Apache.Ignite.Core/Impl/Client/Cache/CacheClient.cs

```diff
@@ -754,14 +797,26 @@ private void WriteRequest(Action<ClientRequestContext> writeAction, ClientReques
         {
             ctx.Stream.WriteInt(_id);
 
+            var flags = ClientCacheRequestFlag.None;
             if (_expiryPolicy != null)
             {
                 ctx.Features.ValidateWithExpiryPolicyFlag();
-                ctx.Stream.WriteByte((byte) ClientCacheRequestFlag.WithExpiryPolicy);
-                ExpiryPolicySerializer.WritePolicy(ctx.Writer, _expiryPolicy);
+                flags = flags | ClientCacheRequestFlag.WithExpiryPolicy;
             }
-            else
-                ctx.Stream.WriteByte((byte) ClientCacheRequestFlag.None); // Flags (skipStore, etc).
+
+            var tx = _ignite.Transactions.CurrentTx;
+            if (tx != null)
+            {
+                flags |= ClientCacheRequestFlag.WithTransactional;
+            }
+
+            ctx.Stream.WriteByte((byte) flags);
+
+            if (flags.HasFlag(ClientCacheRequestFlag.WithExpiryPolicy))
```

Comment:
`HasFlag` is [slow on older frameworks](https://www.code4it.dev/blog/hasflag-performance-benchmarkdotnet), let's avoid it here.